Source/Csla/Core/GraphMerger.cs

```diff
@@ -202,5 +202,173 @@ private void MergeGraph(IEditableCollection target, IEditableCollection source)
         target.Remove(item);
       GetDeletedList<C>(target).Clear();
     }
+
+    #region Async Methods
+    /// <summary>
+    /// Merges state from source graph into target graph.
+    /// </summary>
+    /// <param name="target">Target of merge.</param>
+    /// <param name="source">Source for merge.</param>
+    public async Task MergeGraphAsync(IEditableBusinessObject target, IEditableBusinessObject source)
+    {
+      if (target is IManageProperties imp)
+      {
+        FieldManager.FieldDataManager targetFieldManager = null;
+        if (target is IUseFieldManager iufm)
+          targetFieldManager = iufm.FieldManager;
+        FieldManager.FieldDataManager sourceFieldManager = null;
+        if (source is IUseFieldManager iufms)
+          sourceFieldManager = iufms.FieldManager;
+
+        var targetProperties = imp.GetManagedProperties();
+        foreach (var item in targetProperties)
+        {
+          var sourceFieldExists = true;
+          if (sourceFieldManager != null && (item.RelationshipType & RelationshipTypes.LazyLoad) == RelationshipTypes.LazyLoad)
+            sourceFieldExists = sourceFieldManager.FieldExists(item);
+          object sourceValue = null;
+          if (sourceFieldExists)
+            sourceValue = ReadProperty(source, item);
+          if (sourceValue is IEditableBusinessObject sourceChild)
+          {
+            var targetFieldExists = true;
+            if (targetFieldManager != null && (item.RelationshipType & RelationshipTypes.LazyLoad) == RelationshipTypes.LazyLoad)
+              targetFieldExists = targetFieldManager.FieldExists(item);
+            if (targetFieldExists && ReadProperty(target, item) is IEditableBusinessObject targetChild)
+            {
+              await MergeGraphAsync(targetChild, sourceChild);
+            }
+            else
+            {
+              if ((item.RelationshipType & RelationshipTypes.PrivateField) == RelationshipTypes.PrivateField)
+                Reflection.MethodCaller.CallPropertySetter(target, item.Name, sourceChild);
+              else
+                await Task.Run(() => LoadProperty(target, item, sourceChild));
```

Comment:
Why is there a `Task.Run`? Is there any reason for it? I don't think here should be _any_ `Task.Run` be used. That even hurts performance.

Reply:
noted.

Reply:
Fixed. Please check.